src/dls_bluesky_core/core/maths.py

```diff
@@ -0,0 +1,50 @@
+from typing import Tuple
+
+import numpy as np
+
+
+def step_to_num(start: float, stop: float, step: float) -> Tuple[float, float, int]:
+    """
+    Standard handling for converting from start, stop, step to start, stop, num
+    Forces step to be same direction as length
+    Includes a final point if it is within 1% of the end point (consistent with GDA)
+
+    Args:
+        start (float):
+            Start of length, will be returned unchanged
+        stop (float):
+            End of length, if length/step does not divide cleanly will be returned
+            extended up to 1% of step, or else truncated.
+        step (float):
+            Length of a step along the line formed from start to stop.
+            If stop < start, will be coerced to be backwards.
+
+    Returns:
+        start, truncated_stop, num = Tuple[float, float, int]
+        start will be returned unchanged
+        truncated_stop = start + num * step
+        num is the maximal number of steps that could fit into the length.
+
+    """
+    # Make step be the right direction
+    step = abs(step) if stop >= start else -abs(step)
+    # If stop is within 1% of a step then include it
+    steps = int((stop - start) / step + 0.01)
```

Comment:
This relies on the user of the function not passing in something like start = 1, stop = 10, step = -0.901, when they really mean start = 10, stop = 1, step = 0.901. I know this would strictly be wrong, but people are lazy. Maybe sould include a check to reject or cope with this case (i.e. swap start and stop if step is negative but start is less than stop). If this isn't caught the maths will give the wrongly truncated answer, i.e. 9 steps instead of 10. I'll add a test to demonstrate

Reply:
We *=-1 step if the span is negative but the step is positive, and *=-1 step if the span is positive and the step is negative: either way the span is the "correct" sign and the step is adjusted. 

I think the former is more likely than the latter and I think the handling of the former is sensible, so I think the handling of the latter is also more correct than inverting the span when the step becomes negative.

Reply:
Discussed offline and agreed that current behaviour is correct